labs/lab-12/tasks/playing-god/solution/README.md

```diff
@@ -5,11 +5,29 @@ parent: 'Task: Playing God'
 
 # Solution
 
+## x86-64 (64-bit) Version
+
 Run the executable with GDB.
 You can see the random number in the register before the input function call.
 
+In x86-64, the random number appears in EAX register (lower 32 bits of RAX):
+
 ```asm
-$eax   : 0x12986
-$ebx   : 0x0
-$ecx   : 0x12986
+$rax   : 0x289a27a5
+$eax   : 0x289a27a5
+```
+
+The program generates `rand_r(&seed) % 100000`, so you need to calculate the modulo:
+
+```bash
+python3 -c "print(0x289a27a5 % 100000)" | ./dynamic2
+# Result: 91333
 ```
+
+Or use GDB to extract and calculate:
+
+```bash
+gdb -batch -ex "break *main+67" -ex "run" -ex "print \$eax % 100000" ./dynamic2 < /dev/null
```

Comment:
Because `vuln.c` is not in this folder, the breakpoint at main+67 will generate a warning. I suggest you use the address directly as the binary is linked `no-pie` so the addresses are consistent with the binary at runtime.